### PR TITLE
Automatically skip tests with apigroups which are not served

### PR DESCRIPTION
**Pull request by ingvagabund**
Excluding tests with [apigroup:GROUP] labels for apigroups which are not served by a cluster. E.g. MicroShift is not serving most of the openshift.io apigroups. Other installations might be serving only a subset of the api groups.

jira: https://issues.redhat.com/browse/WRKLDS-488

### CODE REVIEW

File: pkg/test/ginkgo/cmd_runsuite.go [hunk 1]
```diff
@@ -156,6 +156,21 @@ func (opt *Options) Run(suite *TestSuite, junitSuiteName string) error {
 		return err
 	}
 
+	discoveryClient, err := getDiscoveryClient()
```
Comment (deads2k) [inline pkg/test/ginkgo/cmd_runsuite.go:159]: need to find a way for this to not fail when running a --dry-run without a kubeconfig.  I suggest allowing all in such a case.

File: pkg/test/ginkgo/group_filter.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func getDiscoveryClient() (*discovery.DiscoveryClient, error) {
+	cfg := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(clientcmd.NewDefaultClientConfigLoadingRules(), &clientcmd.ConfigOverrides{})
```
Comment (deads2k) [inline pkg/test/ginkgo/group_filter.go:12]: huh, this is how we get our kubeconfig?  Seems weird.

File: pkg/test/ginkgo/cmd_runsuite.go [hunk 3]
```diff
@@ -156,6 +156,21 @@ func (opt *Options) Run(suite *TestSuite, junitSuiteName string) error {
...
+
+	apiGroupFilter, err := newApiGroupFilter(discoveryClient)
+	if err != nil {
+		return fmt.Errorf("unable to build api group filter: %v", err)
+	}
+
+	// Excluding tests with [apigroup:GROUP] labels for apigroups which are not
+	// served by a cluster. E.g. MicroShift is not serving most of the openshift.io
+	// apigroups. Other installations might be serving only a subset of the api groups.
+	tests = apiGroupFilter.filter(tests)
```
Comment (deads2k) [inline pkg/test/ginkgo/cmd_runsuite.go:172]: In my ideal world, I'd get to see the test listed as skipped.  I think this is where that list gets produced https://github.com/openshift/origin/blob/master/pkg/test/ginkgo/junit.go#L30-L40 . Building on this, how much harder is it to produce that?

**Comment by deads2k** [comment]
Note to self, before merge we must see the number of tests being run the same as before this PR.

**Comment by ingvagabund** [comment]
> I think you need something like https://github.com/ingvagabund/origin/pull/4 to actually skip and provide details, right?

Yeap. Thanks for the pointers. I have included your commit in the PR.

**Comment by deads2k** [comment]
1. gcp-csi had the correct number

**Comment by deads2k** [comment]
this is looking good.

/hold cancel

**Pull request merged**